### DISCUSSION TITLE
Best way to approach 1-sided lags

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1504] [General]
As discussed I'd like to add lag support for Linux hosts, and one scenario is like this:
```
links:
- lag:
   members: [ h1-s1, h1-s2 ]
```

i.e. a Linux host connected to 2 switches. In general, it is possible to have a bondX interface on the host side, and nothing special on the switches; the host can choose which link it wants use for active/backup. The switch needn't support the lag module at all for this

Currently, Netlab complains that the switch needs to have lag enabled. I tried setting a ```lag: False``` attribute on the interface, meaning ignore the lag - but that's not allowed either, given that it's an attribute from an unsupported module

A possible resolution could also be "it's a corner case - just require lag support for all nodes". Just asking for opinions

**Comment by ipspace** [comment]
There's no one-sided LAG (by the IEEE definition of LAG?)

Either you implement MLAG on the switches, or you're using some Linux-specific trick on the server side, in which case I don't think we want to know about the details ;)

**Reply by jbemmel** [reply]
For reference: https://www.kernel.org/doc/Documentation/networking/bonding.txt

There are several bonding modes that don't need any special switch side support. We mainly use active/backup, each Linux server uses only 1 link at a time presenting its MAC address. If it needs or wants to, it switches to the backup link

**Reply by ipspace** [reply]
Yeah, I know all about the Linux bonding modes. I wrote an article about them a few years ago ;)

Anyhow, like you probably wouldn't call a tricycle a three-wheeled car, some people wouldn't call Linux bonding 1-sided LAG. Terminology still matters to a few of us, including the writer of the documentation you quoted who only used "aggregation" when referring to 802.1ad (IEEE LAG).

Anyway, I do agree that Linux bonding might be an interesting thing to test, but as it's a single-platform thingy, that should probably be implemented in a plugin, not as a hack in the LAG module.

Simplifying things (and assuming that most sane people would use single bond interface for non-hypervisor hosts), I wouldn't even bother with an abstraction like we have in **lag.members** but just specify some parameter on the Linux interfaces.

**Reply by jbemmel** [reply]
ok, plugin for Linux bonding it is :)
